### ISSUE TITLE
Using Kronecker Transforms on multiple layers 

### ISSUE THREAD

**Issue by arijitthegame** [learnables/learn2learn#262]
Hi, Thank you for this excellent repo. I am using an Adapter Augmented BERT model. I am trying to use ANIL on it. The example for ANIL shows the usage of Kronecker transform on the final classifier layer. However I would like to use fast adaptation on the Adapters. Here is the model definition. 

'''
class AdapterMetaLearner(nn.Module):
    def __init__(self, model, num_labels, dropout_prob, adapter_name = 'pfeiffer', adapter_nonlinearity='gelu', adapter=True):
        """
        Args: model = preloaded BERT model
              num_labels = list of number of labels  
              dropout_prob = dropout probs 
              Added support of adapter modules. In the adapter name pass in either pfeiffer or houlsby. 
              adapter_nonlinearity = str. Currently supports gelu, relu, tanh, sigmoid
        """
       
        super(AdapterMetaLearner, self).__init__()
       
        self.model = model
        self.num_labels = num_labels
        self.dropout_prob = dropout_prob
        self.dropout = nn.Dropout(dropout_prob)
        self.adapter = adapter
        if self.adapter == True: 
            config = AdapterConfig.load(adapter_name, non_linearity=adapter_nonlinearity)
            self.model.add_adapter("metatask", config=config )
            model.active_adapters = "metatask"

        self.meta_classifier = nn.Linear(self.model.config.hidden_size, num_labels)
            
        
    def forward(self,input_ids,attention_mask):

        outputs = self.model(input_ids=input_ids,attention_mask=attention_mask)
        sequence_output = outputs[1]

        sequence_output = self.dropout(sequence_output)
      
        return self.meta_classifier(sequence_output)

'''
In the ANIL example, one sets classifier as the final layer(s). However I want to set classifier as all the adapter layers. The adapters are small bottleneck networks between each BERT output layers. So we add 12 adapters, each with 2 layers. Is there an easy way to use Kronecker transform?

I am talking about the following line of code in the examples folder:
'''
kfo_transform = l2l.optim.transforms.KroneckerTransform(l2l.nn.KroneckerLinear)
fast_update = l2l.optim.ParameterUpdate(
        parameters=classifier.parameters(),
        transform=kfo_transform,
    )
'''
The classifier there is a just a linear linear but here I have a bunch of layers. Sorry about this long convoluted question.

**Comment by seba-1511** [comment]
You need to replace the classifier module with the adapter modules.  You can check the code in [ParameterUpdate](https://github.com/learnables/learn2learn/blob/8de7033a885953a62a34db6ff8cb637d595b904c/learn2learn/optim/parameter_update.py) for how to use Kronecker layers only in the adapters (pass a dictionary of `adapter: kronecker` as the `parameters` argument). Sorry we don't have more detailed docs on this, feel free to reach out on slack.
